File: Python/programmers/etc/level1/예산.py
def solution(d, budget):
    d.sort()
    
    for idx, cost in enumerate(d):
        if budget >= cost:
            budget -= cost
        else:
            return idx
    
    return idx+1


d = [1,3,2,5,4]
budget = 9
print(solution(d, budget)) # 3

d = [2,2,3,3]
budget = 10
print(solution(d, budget)) # 4

File: Java/programmers/etc/level1/예산.java
package CodeTest.Java.programmers.etc.level1;

import java.util.*;

public class 예산 {
    public int solution(int[] costs, int budget) {
        Arrays.sort(costs);
        int answer = 0;
        for (int cost : costs) {
            if (budget >= cost) budget -= cost;
            else break;
            answer++;
        }
        return answer;
    }
}
